src/Security/Group.php

```diff
@@ -377,6 +377,70 @@ public function collateAncestorIDs()
     }
 
     /**
+     * Check if the group is a child of the given group or any parent groups
+     *
+     * @param string|int|Group $group Group instance, Group Code or ID
+     * @return bool Returns TRUE if the Group is a child of the given group, otherwise FALSE
+     */
+    public function inGroup($group)
+    {
+        return in_array($this->identifierToGroupID($group), $this->collateAncestorIDs());
+    }
+
+    /**
+     * Check if the group is a child of the given groups or any parent groups
+     *
+     * @param (string|int|Group)[] $groups
+     * @param bool $requireAll set to TRUE if must be in ALL groups, or FALSE if must be in ANY
+     * @return bool Returns TRUE if the Group is a child of any of the given groups, otherwise FALSE
+     */
+    public function inGroups($groups, $requireAll = false)
+    {
+        if (empty($groups)) {
+            return false;
+        }
+        $ancestorIDs = $this->collateAncestorIDs();
+        $candidateIDs = [];
+        foreach ($groups as $group) {
+            try {
+                $candidateIDs[] = $this->identifierToGroupID($group);
+            } catch (\InvalidArgumentException $e) {
```

Comment:
Going back to @dhensby 's feedback

> I think trying to have a custom error message for every type of invalid argument that's sent in is overkill. It'll be less logic and code just to use a generic message for the exceptions.
> Lets keep code complexity down

It's too hard to get this right just drop exceptions and handle the logic here. If it's $requireAll and null, return false, otherwise if it's null and not $requireAll, don't add it to $candidateIDs. you'll need to move your empty() check to below this loop though, if you could end up with empty $candidateIDs.